arwen/host/arwen.go

```diff
@@ -311,69 +306,55 @@ func (host *vmHost) RunSmartContractCreate(input *vmcommon.ContractCreateInput)
 	host.mutExecution.RLock()
 	defer host.mutExecution.RUnlock()
 
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
+	defer cancel()
+
 	log.Trace("RunSmartContractCreate begin", "len(code)", len(input.ContractCode), "metadata", input.ContractCodeMetadata)
 
-	try := func() {
+	done := make(chan interface{}, 1)
+	go func() {
 		vmOutput = host.doRunSmartContractCreate(input)
-	}
-
-	catch := func(caught error) {
-		err = caught
-		log.Error("RunSmartContractCreate", "error", err)
-	}
+		done <- 1
```

Comment:
not sure if there is a case, but if it is, you could have a defer function inside this go routine that treats any panic that could have occurred (by using recover). this way, we can isolate entirely the sc call execution

Reply:
yes, panic recovery can be added - although it won't happen :dagger: